zone/doors.cpp

```diff
@@ -211,13 +211,23 @@ void Doors::HandleClick(Client* sender, uint8 trigger) {
 
 	uint32 required_key_item       = GetKeyItem();
 	uint8  disable_add_to_key_ring = GetNoKeyring();
-	uint32 player_has_key          = 0;
+	bool player_has_key            = false;
 	uint32 player_key              = 0;
 
 	const EQ::ItemInstance *lock_pick_item = sender->GetInv().GetItem(EQ::invslot::slotCursor);
-	player_has_key = static_cast<uint32>(sender->GetInv().HasItem(required_key_item, 1));
 
-	if (player_has_key != INVALID_INDEX) {
+	// If classic key on cursor rule, check for it, otherwise owning it ok.
+	if (RuleB(Doors, RequireKeyOnCursor)) {
+		if (lock_pick_item != nullptr &&
```

Comment:
Can just do if (lock_pick_item && whatever. Checks for nullptr implicitly.

Reply:
Yeah, once again since they used the check vs nullptr below a few lines  later, I chose to stay consistent with code around me.  If we have a standard to do it one way or the other globally, I'd be on board.